site/app/entities/poll/Poll.php

```diff
@@ -284,4 +301,19 @@ public function getSelectedOptionIds(): array {
         }
         return $selected_options;
     }
+
+    /**
+     * Given a poll and a response, returns whether the response already exists
+     *
+     * @param string $custom_response
+     * @return bool Response in Poll already exists
+     */
+    public function existsCustomResponse(string $custom_response): bool {
+        foreach ($this->getOptions() as $option) {
+            if ($option->getResponse() === $custom_response) {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Surely this could be done more efficiently.  A new method could be created in `OptionRepository.php` for this.

Reply:
Done in [ff1a2f9](https://github.com/Submitty/Submitty/pull/10197/commits/ff1a2f95bc9d73ad04c348d0c7e4458085de2be8). Is this better?